server/game.js

```diff
@@ -109,23 +109,39 @@ export default class Game
 		{
 			// give each player their respective view of what everyone's hand looks like
 			let handsView = playersArray.map((player, i) => {
-				// your own hand
-				if(player.id == playersArray[p].id)
+				// your own hand and you're the dealer
+				if(playersArray[p].gameOwner == true  && player.id == playersArray[p].id)
 					return {
 						playerId: player.id,
 						playerName: player.name,
 						cards: player.cards,
 						currentPlayer: this.#currentPlayerIndex == i,
+						gameOwnerButton: 

```

Comment:
its not great to write html out from javascript, html live in html except in very rare circumstances. Instead of adding markup from js to html when needed, just keep it in the html where it belongs and remove (or hide) it when it isnt needed.

Also youve put this in the server/game.js which is the wrong place for lots of reasons. The playersHands object represents what each players hand looks like to each player. HTML for showing a deal button has nothing to do with this, even if we did want html in js this is a huge mixing of concerns which should be separate.

Reply:
fair point, moved and recommitted to have hide/add functionality reside in view.js, still based on gameOwner property